### PR TITLE
Update example action plugin

### LINKED ISSUE #104350: [CI] Example plugin builds failing due to refactored ActionPlugin
Looks like `ActionPlugin` was refactored in #103277 which is causing compile errors in our example plugin projects.

https://gradle-enterprise.elastic.co/s/bpsse7pry3mek/console-log/task/:rest-handler:compileJava?anchor=4&page=1

### PR DESCRIPTION
**Pull request by thecoop**
Update example plugin for API changes in #103277.

This closes #104350

### CODE REVIEW

**Review by ldematte — APPROVED** [review]
Don't know if you want to wait for es-delivery approval, but LGTM

**Comment by thecoop** [comment]
It's our changes, and a very simple fix - so nothing specific for delivery to look at here

**Pull request merged**
